react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

```diff
@@ -167,15 +176,48 @@ public void onReactContextInitialized(ReactContext context) {
         // The string here (e.g. "MyReactNativeApp") has to match
         // the string in AppRegistry.registerComponent() in index.js
         mReactRootView.setAppProperties(initialProps);
+    }
 
-        if (isNewPost) {
-            setTitle("");
-            setContent("");
+    public void attachToContainer(ViewGroup viewGroup, OnMediaLibraryButtonListener onMediaLibraryButtonListener,
+                                  OnReattachQueryListener onReattachQueryListener) {
+        MutableContextWrapper contextWrapper = (MutableContextWrapper) mReactRootView.getContext();
+        contextWrapper.setBaseContext(viewGroup.getContext());
+
+        mOnMediaLibraryButtonListener = onMediaLibraryButtonListener;
+        mOnReattachQueryListener = onReattachQueryListener;
+
+        if (mReactRootView.getParent() != null) {
+            ((ViewGroup) mReactRootView.getParent()).removeView(mReactRootView);
+        }
+
+        viewGroup.addView(mReactRootView, 0,
+                new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
+
+        refocus();
+    }
+
+    private void refocus() {
+        if (mLastFocusedView != null) {
+            // schedule a request for focus
+            new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
As mentioned in chat convo this could introduce some lag, but in my opinion it's unlikely to happen on "normal" workflow. Even it that happens, the focus/caret is restored properly with just a bit of delay.